velox/functions/sparksql/MapFromArrays.cpp

```diff
@@ -0,0 +1,203 @@
+
+      auto numValues = valuesArray->sizeAt(valueIndices[row]);
+      auto valuesOffset = valuesArray->offsetAt(valueIndices[row]);
+
+      VELOX_USER_CHECK_EQ(
+        numKeys,
+        numValues,
+        "{}",
+        "Key and value arrays must be the same length");
+
+      checkNullsInKey(numKeys, keysOffset, keysElements);
+      auto dedupIndices = deduplicateKeys(numKeys, keysOffset, keysElements);
+      for (auto i = 0; i < numKeys; ++i) {
+        if (dedupIndices.find(keysOffset + i) != dedupIndices.end()) {
+          rawKeysIndices[totalElements] = keysOffset + i;
+          rawValuesIndices[totalElements] = valuesOffset + i;
+          ++totalElements;
+        }
+      }
+
+      rawOffsets[row] = offset;
+      rawSizes[row] = dedupIndices.size();
+      offset += dedupIndices.size();
+    });
+
+    auto wrappedKeys = BaseVector::wrapInDictionary(
+      nullptr, keysIndices, totalElements, keysArray->elements());
+    auto wrappedValues = BaseVector::wrapInDictionary(
+      nullptr, valuesIndices, totalElements, valuesArray->elements());
+    auto mapVector = std::make_shared<MapVector>(
+        context.pool(),
+        outputType,
+        nullptr,
+        rows.end(),
+        offsets,
+        sizes,
+        wrappedKeys,
+        wrappedValues);
+    context.moveOrCopyResult(mapVector, rows, result);
+  }
+
+ private:
+  void checkNullsInKey(
+    vector_size_t size,
+    vector_size_t offset,
+    const VectorPtr& keysElements) const {
+    for (auto i = 0; i < size; ++i) {
+      VELOX_USER_CHECK(
+          !keysElements->isNullAt(offset + i), "map key cannot be null");
+
+      VELOX_USER_CHECK(
+          !keysElements->containsNullAt(offset + i),
+          "{}: {}",
+          "map key cannot be indeterminate",
+          keysElements->toString(offset + i));
+    }
+  }
+
+  std::unordered_set<vector_size_t> deduplicateKeys(
+    vector_size_t numKeys,
+    vector_size_t keysOffset,
+    const VectorPtr& keysElements) const {
+      std::vector<vector_size_t> sortedIndices(numKeys);
+      std::iota(sortedIndices.begin(), sortedIndices.end(), keysOffset);
+      keysElements->sortIndices(sortedIndices, CompareFlags());
+
+      std::unordered_set<vector_size_t> dedupIndices;
+      for (auto i = 0; i < sortedIndices.size()-1; ++i) {
+        auto isDuplicatedKey = keysElements->equalValueAt(
+          keysElements.get(), sortedIndices[i], sortedIndices[i + 1]);
+
+        if constexpr (Policy == core::SparkMapKeyDedupPolicy::EXCEPTION) {
+          if (isDuplicatedKey) {
+            auto duplicateKey = keysElements->wrappedVector()->toString(
+              keysElements->wrappedIndex(sortedIndices[i]));
+            VELOX_USER_FAIL("Duplicate map keys ({}) are not allowed", duplicateKey);
+          }
+        }
+
+        if (!isDuplicatedKey) {
+          dedupIndices.insert(sortedIndices[i]);
+        }
+      }
+      dedupIndices.insert(sortedIndices[sortedIndices.size() - 1]);
```

Comment:
When `numKeys = 0`, `sortedIndices.size() - 1` is `-1`, which can lead to out-of-bounds accesses